rest-client-driver/src/main/java/com/github/restdriver/clientdriver/ClientDriverResponse.java

```diff
@@ -78,71 +87,56 @@ public ClientDriverResponse(String content) {
     }
-    
+
     /**
      * <p>Creates a new response with the given body, a suitable default status
      * code and a given content-type.</p>
      * <p>If the content given is null a 204 status code is given, otherwise 200.</p>
      * 
-     * @param content
-     *            The content of the response
+     * @param stream
+     *            The input stream containing the content of the response
      * @param contentType
      *            The content type
      */
-    public ClientDriverResponse(InputStream content, String contentType) {
-        this(convertInputStreamToByteArray(content), contentType);
-    }
-    
-    private ClientDriverResponse(byte[] content, String contentType) {
-        this.status = statusCodeForContent(content);
-        this.content = content;
-        
-        if (content != null && content.length != 0) {
-            this.contentType = contentType;
-        } else {
-            this.contentType = null;
-        }
-        
-        this.headers = new HashMap<String, String>();
-    }
-    
-    private static byte[] convertStringToByteArray(String content) {
-        return content != null ? content.getBytes() : null;
+    public ClientDriverResponse(InputStream stream, String contentType) {
+        this(DEFAULT_HANDLER, stream, contentType);
     }
-    
-    private static byte[] convertInputStreamToByteArray(InputStream content) {
-        try {
-            return content != null ? IOUtils.toByteArray(content) : null;
-        } catch (IOException e) {
-            throw new ClientDriverResponseCreationException("unable to create client driver response", e);
-        }
+
+    private ClientDriverResponse(Handler handler, InputStream stream, String contentType) {
+        this.status = statusCodeForContent(stream);
+        this.stream = stream;
+        this.contentType = hasBody() ? contentType : null;
+        this.headers = new HashMap<>();
+        this.handler = handler;
     }
     
-    private static int statusCodeForContent(byte[] content) {
+    private static int statusCodeForContent(InputStream content) {
         return content != null ? DEFAULT_STATUS_CODE : EMPTY_RESPONSE_CODE;
     }
     
     /**
      * @return The content as a byte array
+     * @throws IOException if consumption of stream failed
      */
-    public byte[] getContentAsBytes() {
-        if (content == null || content.length == 0) {
+    public byte[] getContentAsBytes() throws IOException {
+        if (stream == null || stream.available() == 0) {
             return null;
         } else {
-            return content;
+            return ByteStreams.toByteArray(stream);
         }
     }
     
     /**
      * @return The content as a string, or an empty string if the content byte array is null or empty.
+     * @throws IOException if consumption of stream failed
      */
-    public String getContent() {
-        if (getContentAsBytes() == null) {
+    public String getContent() throws IOException {
```

Comment:
I think UnchangeIOException would be more appropriate here, this would break many existing testcases.